core/domain/activity_jobs_one_off.py

```diff
@@ -59,6 +52,145 @@ def reduce(key, values):
         pass
 
 
+class AddAllUserIdsOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """For every right model merge the data from all the user id fields
+    together and put them in new all_user_ids field.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Return a list of datastore class references to map over."""
+        return [collection_models.CollectionRightsModel,
+                exp_models.ExplorationRightsModel,
+                topic_models.TopicRightsModel]
+
+    @staticmethod
+    def map(rights_model):
+        """Implements the map function for this job."""
+        commit_message = 'Add new all_user_ids field'
+        commit_cmds = [{'cmd': 'add_new_field', 'field_name': 'all_user_ids'}]
```

Comment:
If this is a structural change, I don't think this should be made as a commit (which will end up creating new historical models). Shouldn't we be surgically updating all the existing versions of all the models in the datastore instead (but not create new ones)?